test/system/projects/samples/attachments_test.rb

```diff
@@ -118,31 +119,31 @@ class AttachmentsTest < ApplicationSystemTestCase
           click_on I18n.t('projects.samples.attachments.attachment.delete'), match: :first
         end
 
-        within('dialog') do
+        within('dialog[open]') do
           assert_accessible
           assert_text I18n.t('projects.samples.attachments.delete_attachment_modal.description')
           click_button I18n.t('projects.samples.attachments.delete_attachment_modal.submit_button')
         end
 
-        assert_text I18n.t('projects.samples.attachments.destroy.success', filename: 'test_file_A.fastq')
+        assert_text I18n.t('projects.samples.attachments.destroy.success', filename: 'test_file_B.fastq')
         within('#table-listing') do
-          assert_no_text 'test_file_A.fastq'
+          assert_no_text 'test_file_B.fastq'
         end
       end
 
       test 'user with role >= Maintainer should be able to attach, view, and destroy paired files to a Sample' do
         visit namespace_project_sample_url(@namespace, @project, @sample2)
         # Initial View
-        assert_selector 'a', text: I18n.t('projects.samples.show.new_attachment_button'), count: 1
+        assert_selector 'a', text: I18n.t('projects.samples.show.new_attachment_button')
         within('#table-listing') do
           assert_text I18n.t('projects.samples.attachments.table.empty_state.title')
           assert_text I18n.t('projects.samples.attachments.table.empty_state.description')
-          assert_selector 'button', text: I18n.t('projects.samples.attachments.attachment.delete'), count: 0
+          assert_selector 'button', text: I18n.t('projects.samples.attachments.attachment.delete'), visible: false
```

Comment:
Same comment as above to use `assert_selector 'button[disabled]', text: BUTTON_TEXT_HERE`